dashboard/src/pages/TreeDetails/TreeDetails.tsx

```diff
@@ -300,22 +261,25 @@ function TreeDetails(): JSX.Element {
         />
       </div>
       <QuerySwitcher
-        status={testStatus}
-        data={testsData}
+        status={status}
+        data={data}
         skeletonClassname="h-[200px] bg-lightGray"
       >
         <div className="mt-5">
           <MemoizedHardwareUsed
             title={<FormattedMessage id="treeDetails.hardwareUsed" />}
-            hardwareUsed={testsData?.hardwareUsed}
+            hardwareUsed={data?.hardwareUsed}
             diffFilter={diffFilter}
           />
         </div>
       </QuerySwitcher>
       <div className="flex flex-col pb-2">
         <div className="sticky top-[4.5rem] z-10">
           <div className="absolute right-0 top-2 py-4">
-            <TreeDetailsFilter paramFilter={diffFilter} treeUrl={treeUrl} />
+            <TreeDetailsFilter
+              paramFilter={diffFilter}
+              treeUrl={data?.treeUrl ?? ''}
+            />
           </div>
```

Comment:
This probably should not exist without treeUrl

```suggestion         
{ data?.treeUrl ( <TreeDetailsFilter
              paramFilter={diffFilter}
              treeUrl={data?.treeUrl ?? ''}
            />
          </div>)
```